### PR TITLE
LUT

### PR DESCRIPTION
**Pull request by kelvinlau20100**

### CODE REVIEW

File: src/subclient.ts [hunk 1]
```diff
@@ -1542,12 +1573,14 @@ export class SubClient {
     this.delegatedCheck();
     let tx = this.getPositionMovementTx(movementType, movements);
     let response = await utils.simulateTransaction(this._parent.provider, tx);
-
-    let events = response.events;
     let positionMovementEvent = undefined;
-    for (var i = 0; i < events.length; i++) {
-      if (events[i].name == "PositionMovementEvent") {
-        positionMovementEvent = events[i].data;
+    while (true) {
```
Comment (filipzeta) [inline src/subclient.ts:1577]: whats the purpose of this change?

File: src/utils.ts [hunk 2]
```diff
@@ -691,43 +690,63 @@ export async function processTransaction(
...
+    }
 
-  if (useLedger) {
-    tx = await Exchange.ledgerWallet.signTransaction(tx);
+    let v0Tx: VersionedTransaction = new VersionedTransaction(
+      new TransactionMessage({
+        payerKey: provider.wallet.publicKey,
+        recentBlockhash:
+          blockhash ??
+          (await provider.connection.getRecentBlockhash()).blockhash,
```
Comment (filipzeta) [inline src/utils.ts:708]: getRecentBlockhash should change to `getLatestBlockhash(commitmentConfig("finalized"))` as per https://github.com/zetamarkets/sdk/pull/193

File: src/utils.ts [hunk 3]
```diff
@@ -691,43 +690,63 @@ export async function processTransaction(
...
+        })
+    );
+    v0Tx = (await provider.wallet.signTransaction(
+      v0Tx
+    )) as VersionedTransaction;
+    rawTx = v0Tx.serialize();
   } else {
-    tx = await provider.wallet.signTransaction(tx);
+    tx.recentBlockhash =
+      blockhash ?? (await provider.connection.getRecentBlockhash()).blockhash;
```
Comment (filipzeta) [inline src/utils.ts:725]: finalized blockhash here too pls :)

File: src/subclient.ts (hunk 1 above)
Comment (kelvinlau20100) [reply src/subclient.ts:1577]: new version of anchor has changed the event parsing

File: src/utils.ts (hunk 2 above)
Comment (kelvinlau20100) [reply src/utils.ts:708]: yep apologies

**Comment by xjcaa** [comment]
what's the impact of https://github.com/coral-xyz/anchor/pull/2018/files?

File: src/program-types.ts [hunk 4]
```diff
@@ -145,7 +145,7 @@ export interface Position {
 
 export interface OrderState {
   closingOrders: anchor.BN;
-  openingOrders: [anchor.BN, anchor.BN];
+  openingOrders: Array<anchor.BN>;
```
Comment (xjcaa) [inline src/program-types.ts:148]: was there an error for this

File: src/utils.ts [hunk 5]
```diff
@@ -726,19 +705,67 @@ export async function processTransaction(
     );
   }
 
-  if (useLedger) {
-    tx = await Exchange.ledgerWallet.signTransaction(tx);
+  if (useVersioned) {
```
Comment (xjcaa) [inline src/utils.ts:708]: is there a reason to useVersioned apart from LUT? - if not, could we assume if LUT is not undefined then we use versioned?

**Review by xjcaa — APPROVED** [review]

File: src/program-types.ts (hunk 4 above)
Comment (kelvinlau20100) [reply src/program-types.ts:148]: yes

File: src/utils.ts (hunk 5 above)
Comment (kelvinlau20100) [reply src/utils.ts:708]: yes i think that makes sense

File: src/utils.ts [hunk 6]
```diff
@@ -692,31 +692,10 @@ export async function processTransaction(
   signers?: Array<Signer>,
   opts?: ConfirmOptions,
   useLedger: boolean = false,
+  lutAccs?: AddressLookupTableAccount[],
```
Comment (xjcaa) [inline src/utils.ts:695]: default to undefined?

Comment (kelvinlau20100) [reply src/utils.ts:695]: can't default value it because it's an optional argument

**Pull request merged**